.github/workflows/publish.yaml

```diff
@@ -0,0 +1,55 @@
+name: Publish ROCK
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yaml
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
```

Comment:
If this can timeout I guess you want this to be manually triggerable, no? I understand from the top of the file that it will only trigger on changes to `main`. What happens if it times out, does it try again?

Reply:
If it times out I can manually re-run that action via `Actions` - there is an option to re-run failed workflows with debugging enabled.